src/Utils.js

```diff
@@ -80,7 +81,12 @@ export function nodeEqual(a, b) {
       return false;
     }
   }
-  return leftKeys.length === Object.keys(right).length;
+
+  if (typeof a !== 'string' && typeof a !== 'number') {
+    return leftKeys.length === Object.keys(right).length;
```

Comment:
I suppose that since react requires ES5 and above, it is safe to conclude that at least an es5-shim will be loaded in any environment enzyme will be used...